designer/server/src/main/scala/pl/touk/nussknacker/ui/limits/LimitsService.scala

```diff
@@ -0,0 +1,158 @@
+package pl.touk.nussknacker.ui.limits
+
+import cats.data.EitherT
+import cats.effect.IO
+import cats.effect.std.Mutex
+import cats.effect.unsafe.IORuntime
+import com.typesafe.scalalogging.LazyLogging
+import pl.touk.nussknacker.engine.ProcessingTypeConfig.LimitsConfig
+import pl.touk.nussknacker.engine.api.deployment.DeploymentUpdateStrategy
+import pl.touk.nussknacker.engine.api.deployment.DeploymentUpdateStrategy.{
+  DontReplaceDeployment,
+  ReplaceDeploymentWithSameScenarioName
+}
+import pl.touk.nussknacker.engine.api.process.{ProcessingType, ProcessName}
+import pl.touk.nussknacker.ui.limits.LimitsService.LimitError
+import pl.touk.nussknacker.ui.limits.LimitsService.LimitError.ActiveScenariosLimitExceededError
+import pl.touk.nussknacker.ui.process.deployment.scenariostatus.ScenarioStatusProvider
+import pl.touk.nussknacker.ui.process.processingtype.provider.ProcessingTypeDataProvider
+import pl.touk.nussknacker.ui.security.api.{LoggedUser, NussknackerInternalUser}
+
+class LimitsService(
+    globalLimitsConfig: GlobalLimitsConfig,
+    activeScenariosLimitProvider: ProcessingTypeDataProvider[LimitsConfig, _],
+    scenarioStatusProvider: ScenarioStatusProvider
+) extends LazyLogging {
+
+  def checkScenarioLimitsBeforeDeploymentUnsafe(
```

Comment:
Maybe let's call it `...NoCrititicalSection` instead of `...Unsafe`. I thought that it had non-monadic return type :)